### ISSUE TITLE
Resolving a service registered in a framework fails if Swinject is integrated via Swift Package Manager

### ISSUE THREAD

**Issue by chkpnt** [Swinject/Swinject#425]
I'm just playing around with the integration of Swinject using Swift Package Manager in Xcode 11 beta 4 and observing an issue I can't reproduce if Swinject is integrated with Carthage.

I have an App-Target as well as a Framework-target, both using Swinject. The framework registers its service `SomeFrameworkStuff` to the app's container using an assembly. Logging the container's content shows that the registration succeeded:
```
Available through Swinject Container: [
    { Service: Bla, Factory: Resolver -> Bla, ObjectScope: graph },
    { Service: SomeFrameworkStuff, Factory: Resolver -> SomeFrameworkStuff, ObjectScope: graph }
]
```
But when I try to resolve `SomeFrameworkStuff.self`, I get
```
Swinject: Resolution failed. Expected registration:
	{ Service: SomeFrameworkStuff, Factory: Resolver -> SomeFrameworkStuff }
```

I have various duplicate class warnings, too:
```
[...]
objc[34972]: Class _TtC8Swinject8SpinLock is implemented in both /Users/ich/Library/Developer/Xcode/DerivedData/SwinjectDemo-dujsuddxiashuwezjwqwxafkvchr/Build/Products/Debug-iphonesimulator/SomeFramework.framework/SomeFramework (0x102ea18e8) and /Users/ich/Library/Developer/CoreSimulator/Devices/2CAB296B-2A4B-47A7-9E76-682BA1E6FA4A/data/Containers/Bundle/Application/7D3C3CCC-67BC-44F1-A57C-769083060488/SwinjectDemo.app/SwinjectDemo (0x102b08520). One of the two will be used. Which one is undefined.
objc[34972]: Class _TtC8Swinject20SynchronizedResolver is implemented in both /Users/ich/Library/Developer/Xcode/DerivedData/SwinjectDemo-dujsuddxiashuwezjwqwxafkvchr/Build/Products/Debug-iphonesimulator/SomeFramework.framework/SomeFramework (0x102ea1980) and /Users/ich/Library/Developer/CoreSimulator/Devices/2CAB296B-2A4B-47A7-9E76-682BA1E6FA4A/data/Containers/Bundle/Application/7D3C3CCC-67BC-44F1-A57C-769083060488/SwinjectDemo.app/SwinjectDemo (0x102b085b8). One of the two will be used. Which one is undefined.
objc[34972]: Class _TtGCs18_DictionaryStorageV8Swinject10ServiceKeyPS0_20ServiceEntryProtocol__$ is implemented in both /Applications/Xcode-11-beta4.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/usr/lib/swift/libswiftCore.dylib (0x7fff899aeb78) and /Applications/Xcode-11-beta4.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/
[...]
objc[34972]: Class _TtGC8Swinject12ServiceEntryP13SomeFramework18SomeFrameworkStuff__ is implemented in both /Applications/Xcode-11-beta4.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/usr/lib/swift/libswiftCore.dylib (0x7fff899aeda8) and /Applications/Xcode-11-beta4.app/Contents/Developer/Platforms/iPhoneOS.platform/Library/Developer/CoreSimulator/Profiles/Runtimes/iOS.simruntime/Contents/Resources/RuntimeRoot/usr/lib/swift/libswiftCore.dylib (0x7fff899b49f0). One of the two will be used. Which one is undefined.
```

The Xcode project demonstrating the issue is available at [chkpnt/SwinjectDemo](https://github.com/chkpnt/SwinjectDemo).

**Comment by jakubvano** [comment]
Looks like this is an issue with linking static libs to dynamic frameworks. If I remove Swinject from `SwinjectDemo`'s link with libraries build phase, it works OK.

**Comment by chkpnt** [comment]
I can't compile the framework if Swinject is removed from the build phase. Is there a step I'm missing?

**Comment by jakubvano** [comment]
I removed it only from the `SwinjectDemo` target, kept it in `SwinjectFramework` build phases.

**Comment by chkpnt** [comment]
Okay, the static lib is indeed causing this issue. Your workaround works, but only if there is not more than one framework involved. (I've added a second framwork to my demonstrator).

The only way I've found Xcode to build Swinject as a dynamic library is by modifying its `Package.swift` to
```swift
   ...
    products: [
        .library(name: "Swinject", type: .dynamic, targets: ["Swinject"])
    ],
   ...
```
This way, everything is working as expected. But in my opinion, this shouldn't be needed. In the documentation of SPM, the following [is written](https://github.com/apple/swift-package-manager/blob/master/Documentation/PackageDescription.md#methods-2):
> A library's product can either be statically or dynamically linked. It is recommended to not declare the type of library explicitly to let the Swift Package Manager choose between static or dynamic linking depending on the consumer of the package.

-> There should be an option in Xcode to define how to consume the package.
